tests/integration/targets/postgresql_query/tasks/postgresql_query_initial.yml

```diff
@@ -41,8 +41,8 @@
       - result.query_list == ['ANALYZE test_table']
       - result.rowcount == 0
       - result.statusmessage == 'ANALYZE'
-      - result.query_result == {}
-      - result.query_all_results == [{}]
+      - result.query_result == {} or result.query_result == []  # Psycopg2 or Psycopg 3
+      - result.query_all_results == [{}] or result.query_all_results == [[]]  # Psycopg2 or Psycopg 3
```

Comment:
Do you think if it's impossible to convert returned rows to dictionaries in Psycopg 3?
```
query_result:
    description:
    - List of dictionaries in column:value form representing returned rows.
    - When running queries from a file, returns result of the last query.
    returned: success
    type: list
    elements: dict
    sample: [{"Column": "Value1"},{"Column": "Value2"}]
```

Reply:
asking as it'd be a breaking change as users can rely on checking if the result is empty or not

Reply:
so i would suggest converting

Reply:
I think both versions return list of dictionaries, when there is data to return.
The only difference in behaviour is when there is no result set.

What should be returned in query_result when there is nothing to return? Empty list or empty dictionary? 
Using Psycopg 3 we get an empty list `[]` which may be the correct option.

The options would be to 
a) fix the existing behaviour 
b) keep the existing behaviour 
c) accept existing behaviour but fix it for psycopg3.

Reply:
i think it should continue to return what it returns now when empty, i.e. `[{}]`, until there are reasonable concerns raised
my opinion:)

Reply:
ok, lets keep the existing behaviour